packages/react-performance/README.md

```diff
@@ -0,0 +1,261 @@
+
+### Reporting data to the server
+
+Performance metrics data is most useful when it can be aggregated and used to make smart optimization decisions or warn you when your app is beginning to become slow. To aggregate such data you'll need to be able to send it to a server. `@shopify/react-performance` provides the `PerformanceReport` component to facilitate this.
+
+> This section is a work in progress. It will be filled out as helper libraries for marshalling data on the server are added to Quilt.
+
+## API
+
+Most APIs provided by this package are available as both hooks (ie. `useNavigationListener`), as well as components (ie. `<NavigationListener />`).
+
+This package also re-exports all of the API of [`@shopify/performance`](https://www.npmjs.com/package/@shopify/performance).
+
+### Hooks
+
+#### useLifecycleEventListener
+
+A custom hook which takes in callback to invoke whenever the `Performance` context object emits a lifecycleEvent.
+
+```tsx
+import {useLifecycleEventListener} from '@shopify/react-performance';
+
+function SomeComponent() {
+  useLifecycleEventListener(event => {
+    console.log(event);
+  });
+
+  return <div>something</div>;
+}
+```
+
+#### useNavigationListener
+
+A custom hook which takes in a callback to invoke whenever a navigation takes place, whether it is a full-page load (such as the result of `location.assign`) or an incrmental load (such as the result of `router.push` in a ReactRouter app).
+
+```tsx
+import {useNavigationListener} from '@shopify/react-performance';
+
+function SomeComponent() {
+  useNavigationListener(navigation => {
+    console.log(navigation);
+  });
+
+  return <div>something</div>;
+}
+```
+
+#### usePerformanceMark
+
+A custom hook which takes an `id` and `stage` and uses them to generate a tag for a call to `window.performance.mark` when the component is mounted. This can be used to mark specialized moments in your applications startup, or a specific interaction.
+
+```tsx
+import React from 'react';
+import {usePerformanceMark} from '@shopify/react-performance';
+
+function SomeComponent() {
+  usePerformanceMark('some-stage', 'some-id');
+
+  return <div>something</div>;
+}
+```
+
+If the hook is called with a `stage` of `usable` or `complete`, it also calls the appropriate method on the `Performance` object from context, allowing you to trigger the `finish` and `usable` lifecycleEvents based on a particular component mounting.
+
+```tsx
+import React from 'react';
+import {usePerformanceMark} from '@shopify/react-performance';
+
+function ProductPage() {
+  // this will call `performance.finish()` on the Performance object in context
+  usePerformanceMark('complete', 'products');
+
+  return <div>cool product page</div>;
+}
+```
+
+#### usePerformanceReport
+
+> Todo
+
+### Components
+
+#### LifecycleEventListener
```

Comment:
Maybe the doc instead of showing each individual API, should explain each concept first and then mention you can use it as a component or a hook.

Possible outline:

### Lifecycle Event Listener
Emit an `onEvent` callback whenever a lifecycle event occur.
The list of  lifecycle event is as follow:
1. `TimeToFirstByteEvent`: The time until the server sends the first part of response. [Learn more](https://developers.google.com/web/tools/chrome-devtools/network/understanding-resource-timing)
2.......

Should be place near the top of the react tree.
Only need to be include once in an application.
Can use component or hook.
See `Performance Report` below for how to compose it together.

Reply:
I think we definitely need to add docs about what exact lifecycle events there are, but probably they belong to `@shopify/performance` and then `@shopify/react-performance` should link out. 

I will definitely do a subsequent PR to flesh out the docs in the base library though 👍